sympytorch/sympy_module.py

```diff
@@ -66,25 +66,29 @@ def fn_(*args):
 class _Node(torch.nn.Module):
     def __init__(self, *, expr, _memodict, _func_lookup, **kwargs):
         super().__init__(**kwargs)
-
+        
         self._sympy_func = expr.func
-
         if issubclass(expr.func, sympy.Float):
             self._value = torch.nn.Parameter(torch.tensor(float(expr)))
             self._torch_func = lambda: self._value
             self._args = ()
-        elif issubclass(expr.func, sympy.UnevaluatedExpr):
-            if len(expr.args) != 1 or not issubclass(expr.args[0].func, sympy.Float):
-                raise ValueError("UnevaluatedExpr should only be used to wrap floats.")
-            self.register_buffer('_value', torch.tensor(float(expr.args[0])))
-            self._torch_func = lambda: self._value
+        elif issubclass(expr.func, sympy.Rational):
+            self.register_buffer('_numerator', torch.tensor(expr.p, dtype=torch.float64))
+            self.register_buffer('_denominator', torch.tensor(expr.q, dtype=torch.float64))
+            self._torch_func = lambda: self._numerator / self._denominator
             self._args = ()
         elif issubclass(expr.func, sympy.Integer):
             # Can get here if expr is one of the Integer special cases,
             # e.g. NegativeOne
             self._value = int(expr)
             self._torch_func = lambda: self._value
             self._args = ()
+        elif issubclass(expr.func, sympy.UnevaluatedExpr):
+            if len(expr.args) != 1 or not issubclass(expr.args[0].func, sympy.Float):
+                raise ValueError("UnevaluatedExpr should only be used to wrap floats.")
+            self.register_buffer('_value', torch.tensor(float(expr.args[0])))
+            self._torch_func = lambda: self._value
+            self._args = ()
```

Comment:
Is there a reason this block of code got moved?

Reply:
will put back in place